src/Compiler.Dynamic/SystemWebCompilation.cs

```diff
@@ -122,11 +124,52 @@ private IEnumerable<BaseTemplateParser> GetParsersToCompile(IEnumerable<string>
             if (ReferenceEquals(stack.Peek(), currentPath))
             {
                 stack.Pop();
+                parser.Parse();
                 yield return parser;
             }
         }
     }
 
+    private IEnumerable<string> GetDependencyPaths(TemplateParser parser)
+    {
+        if (!_dependencies.TryGetValue(parser.CurrentVirtualPathString, out var dependentPaths))
```

Comment:
where is this _dependencies coming from? That's state that I don't want in the general systemwebcompilation class - it's state that is specific to the current compilation

Reply:
resolving as things are simplified now